### PR TITLE
improve filtered search combining title and artist

### PR DESCRIPTION
**Pull request by ntruchsess**
In spotify when browsing the library you can go by artist, then you get individual titles but not the albums.
Or you go by album, then can filter by title. 
So far there is no way to filter your favorits for albums of a specific artist.

This change finds both the titles and the artists in the filtered view.

### CODE REVIEW

File: app/src/main/java/me/hufman/androidautoidrive/carapp/music/views/FilterInputView.kt [hunk 1]
```diff
@@ -50,15 +50,16 @@ class FilterInputView(val state: RHMIState,
 					// still loading
 					sendSuggestions(listOf(FILTERRESULT_LOADING))
 				} else {
-					val suggestions = musicList.asSequence().filter {
-						UnicodeCleaner.clean(it.title ?: "").split(Regex("\\s+")).any { word ->
-							word.toLowerCase(Locale.ROOT).startsWith(input.toLowerCase(Locale.ROOT))
+					val results = input.toLowerCase(Locale.ROOT).let { inputLower ->
```
Comment (hufman) [inline app/src/main/java/me/hufman/androidautoidrive/carapp/music/views/FilterInputView.kt:53]: Good optimization, to do `input.toLowerCase` first!

**Comment by hufman** [comment]
I like this idea! I had originally only considered filtering the list as just a simple set of strings, but it definitely makes sense to also filter with the second-line Artist information!
The current code had two Sequences so that it would return results that have words starting with the query first, and then any results that include the query anywhere within them. The failing unit test checks this feature: When searching for "n", the suggestion "New song" should be above "Best snew song". Can you include your new feature within this two-sequence approach? Something like:
```
filter {
  UnicodeCleaner.clean(it.title ?: "").split(Regex("\\s+")).any { word ->
word.toLowerCase(Locale.ROOT).startsWith(inputLower) } ||
  UnicodeCleaner.clean(it.artist ?: "").split(Regex("\\s+")).any { word ->
word.toLowerCase(Locale.ROOT).startsWith(inputLower) }
} +
filter { .. contains }

**Comment by ntruchsess** [comment]
makes sense. Changed accordingly. Actually there is no need to create the sequence twice, one can just reuse it. As it is a sequence it is evaluated late so the whole search will execute the UnicodeCleaner etc... 15 times (as of take-statement) at max.

**Comment by hufman** [comment]
Oh interesting! Coming from a Python background, where generator functions can only emit elements a single time, I just assumed sequences have the same one-time-use limitation.
Thank you so much!

**Pull request merged**